Remove NVC without deprecation phase (#1694)
cocotb/cocotb: documentation/source/newsfragments/1693.removal.rst

> Makefile and documentation for the NVC simulator which hasn't been working since some time have been removed.

Reviewer: Not really necessary but ...
```suggestion
Makefile and documentation for the NVC simulator which does not work (and never was) have been removed.
```

Author: Done in this spirit.

> Makefile and documentation for the NVC simulator which hasn't been working since some time have been removed.

Reviewer 2: ```suggestion
Support for the NVC simulator has been removed.
```

Author: I initially had it like that, but this sounds as if there ever was "support" to speak of in the first place, which apparently there wasn't. "Non-support has been  removed" would be correct :)
